src/render/buffer/textatlas.js

```diff
@@ -76,7 +76,7 @@ export class TextAtlas extends Atlas {
     const quote = (str) => `${str.replace(/(['"\\])/g, "\\$1")}`;
     const font = this.font.map(quote).join(", ");
 
-    const context = canvas.getContext("2d");
+    const context = canvas.getContext("2d", { willReadFrequently: true });
```

Comment:
_As part of reviewing this, I was reading a bit about this option and want to leave some notes here for our future selves._

[MDN says](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/getContext): 
> `willReadFrequently `: A boolean value that indicates whether or not a lot of read-back operations are planned. This will force the use of a software (instead of hardware accelerated) 2D canvas and can save memory when calling [getImageData()](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/getImageData) frequently.

- "...will force the use of software (instead of hardware accelerated) ..." initially made me a bit wary, but this is a separate `canvas` element than the main 3d scene uses. As far as I can tell, this `canvas` element is just a throw-away canvas used for style computations/measurements.
- Regarding "...when calling [getImageData()](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/getImageData) frequently". Worth mentioning that I doubt this warning was really relevant. I think `getImageData` only gets called once per text value. So if you have 3 axis labels, it'll only get called 3 times (until the labels are changed).

All in all, I suspect this warning wasn't really relevant. But...irrelevant warnings are noise, and I don't think turning setting this to true should cause any problems, so 🚀

Reply:
Agree on the final point! I added a bunch of axis labels on this demo: https://sicmutils.github.io/sicmutils-viewers/src/phase_portrait.html which is what was triggering this in the first place.